mlrun/runtimes/pod.py

```diff
@@ -198,6 +213,124 @@ def _set_volume_mount(self, volume_mount):
         volume_mount_key = hash(f"{volume_name}-{volume_sub_path}-{volume_mount_path}")
         self._volume_mounts[volume_mount_key] = volume_mount
 
+    def _verify_and_set_limits(
+        self,
+        resources_field_name,
+        mem=None,
+        cpu=None,
+        gpus=None,
+        gpu_type="nvidia.com/gpu",
+        only_verify=False,
+    ):
+        if mem:
+            verify_field_regex(
+                f"function.spec.{resources_field_name}.limits.memory",
+                mem,
+                mlrun.utils.regex.k8s_resource_quantity_regex,
+            )
+        if cpu:
+            verify_field_regex(
+                f"function.spec.{resources_field_name}.limits.cpu",
+                cpu,
+                mlrun.utils.regex.k8s_resource_quantity_regex,
+            )
+        # https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/
+        if gpus:
+            verify_field_regex(
+                f"function.spec.{resources_field_name}.limits.gpus",
+                gpus,
+                mlrun.utils.regex.k8s_resource_quantity_regex,
+            )
+        resources = generate_resources(mem=mem, cpu=cpu, gpus=gpus, gpu_type=gpu_type)
+        if only_verify:
+            return resources
```

Comment:
Simply add a function `_verify_limits` and call it from here, no need for the `only_verify` param